src/plugins/intel_gna/src/transformations/matops_to_dwsc.cpp

```diff
@@ -0,0 +1,261 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "matops_to_dwsc.hpp"
+
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/pass/graph_rewrite.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+#include <openvino/cc/ngraph/itt.hpp>
+
+namespace ov {
+namespace intel_gna {
+namespace pass {
+
+// Check if the previous node is convolution, groupconvolution or matmul followed by add
+static bool IsFusable(const Output<Node>& parent) {
+    if (nullptr == std::dynamic_pointer_cast<ngraph::opset1::GroupConvolution>(parent.get_node()->shared_from_this()) ||
+        nullptr == std::dynamic_pointer_cast<ngraph::opset1::Convolution>(parent.get_node()->shared_from_this()) ||
+        nullptr == std::dynamic_pointer_cast<ngraph::opset1::MatMul>(parent.get_node()->shared_from_this()))
+        return false;
+    return true;
+}
+
+static bool Decompose(std::shared_ptr<ov::Node> math_node) {
```

Comment:
Would you consider breaking this function down into a few smaller ones? It would make the code easier to read and maintain.

Reply:
Done